File: app/src/main/java/com/example/xianweili/newsapp/data/service/NewsListsService.java
package com.example.xianweili.newsapp.data.service;

import com.example.xianweili.newsapp.data.model.responsemodel.NewsListsResponse;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by xianweili on 7/24/18.
 */

public interface NewsListsService {
    @GET("/top-headlines")
    Observable<NewsListsResponse> getNewsList(@Query("country") String country,
                                               @Query("apiKey") String apiKey,
                                               @Query("page") int page,
                                               @Query("pagesize") int pagesize);
}

File: app/src/main/java/com/example/xianweili/newsapp/data/service/NewsListsService.kt
package com.example.xianweili.newsapp.data.service

import retrofit2.http.GET
import com.example.xianweili.newsapp.data.model.responsemodel.NewsListsResponse
import io.reactivex.Observable
import retrofit2.http.Query

interface NewsListsService {
    @GET("v2/top-headlines")
    fun getNewsList(
        @Query("country") country: String?,
        @Query("apiKey") apiKey: String?,
        @Query("page") page: Int,
        @Query("pageSize") pagesize: Int
    ): Observable<NewsListsResponse?>?
}